### DISCUSSION TITLE
Server crash  continuously

### DISCUSSION THREAD

**Discussion by celophys** [FabricMC/fabric-api#2027] [Player support]
The crash log:

Time: 2/23/22, 8:18 PM
Description: Ticking entity

java.lang.ClassCastException: class net.minecraft.class_3989 cannot be cast to class net.minecraft.class_1657 (net.minecraft.class_3989 and net.minecraft.class_1657 are in unnamed module of loader net.fabricmc.loader.impl.launch.knot.KnotClassLoader @5f341870)
	at net.minecraft.item.ItemStack.handler$znb000$finishUsingItem(ItemStack:1141)
	at net.minecraft.item.ItemStack.finishUsing(ItemStack)
	at net.minecraft.entity.LivingEntity.consumeItem(LivingEntity:3079)
	at net.minecraft.entity.LivingEntity.tickItemStackUsage(LivingEntity:2959)
	at net.minecraft.entity.LivingEntity.tickActiveItemStack(LivingEntity:2946)
	at net.minecraft.entity.LivingEntity.tick(LivingEntity:2312)
	at net.minecraft.entity.mob.MobEntity.tick(MobEntity:345)
	at net.minecraft.server.world.ServerWorld.tickEntity(ServerWorld:699)
	at net.minecraft.world.World.tickEntity(World:487)
	at net.minecraft.server.world.ServerWorld.method_31420(ServerWorld:360)
	at net.minecraft.world.EntityList.forEach(EntityList:54)
	at net.minecraft.server.world.ServerWorld.tick(ServerWorld:328)
	at net.minecraft.server.MinecraftServer.tickWorlds(MinecraftServer:879)
	at net.minecraft.server.dedicated.MinecraftDedicatedServer.tickWorlds(MinecraftDedicatedServer:324)
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer:823)
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer:684)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer:270)
	at java.lang.Thread.run(Thread.java:833)

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at net.minecraft.class_1799.handler$znb000$finishUsingItem(class_1799.java:1141)
	at net.minecraft.class_1799.method_7910(class_1799.java)
	at net.minecraft.class_1309.method_6040(class_1309.java:3079)
	at net.minecraft.class_1309.method_37119(class_1309.java:2959)
	at net.minecraft.class_1309.method_6076(class_1309.java:2946)
	at net.minecraft.class_1309.method_5773(class_1309.java:2312)
	at net.minecraft.class_1308.method_5773(class_1308.java:345)
	at net.minecraft.class_3218.method_18762(class_3218.java:699)
	at net.minecraft.class_1937.method_18472(class_1937.java:487)
	at net.minecraft.class_3218.method_31420(class_3218.java:360)
	at net.minecraft.class_5574.method_31791(class_5574.java:54)
	at net.minecraft.class_3218.method_18765(class_3218.java:328)

-- Entity being ticked --
Details:
	Entity Type: minecraft:wandering_trader (net.minecraft.class_3989)
	Entity ID: 1099
	Entity Name: Wandering Trader
	Entity's Exact location: -83.50, 68.00, 14.50
	Entity's Block location: World: (-84,68,14), Section: (at 12,4,14 in -6,4,0; chunk contains blocks -96,-64,0 to -81,319,15), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,-64,0 to -1,319,511)
	Entity's Momentum: 0.00, -0.08, 0.00
	Entity's Passengers: []
	Entity's Vehicle: null
	Entity NBT: {AbsorptionAmount:0.0f,ActiveEffects:[{Ambient:0b,Amplifier:0b,Duration:3600,Id:14b,ShowIcon:1b,ShowParticles:1b}],Age:0,Air:300s,ArmorDropChances:[0.085f,0.085f,0.085f,0.085f],ArmorItems:[{},{},{},{}],Attributes:[{Base:16.0d,Modifiers:[{Amount:7.186874414237064E-4d,Name:"Random spawn bonus",Operation:1,UUID:[I;764212988,-1642705748,-1525631181,-1010795249]}],Name:"minecraft:generic.follow_range"},{Base:0.699999988079071d,Name:"minecraft:generic.movement_speed"}],BMLootBlock:0b,Brain:{memories:{}},CanPickUpLoot:0b,DeathTime:0s,DespawnDelay:0,FallDistance:0.0f,FallFlying:0b,Fire:-1s,ForcedAge:0,HandDropChances:[0.085f,0.085f],HandItems:[{Count:1b,id:"minecraft:potion",tag:{Potion:"minecraft:invisibility"}},{}],Health:20.0f,HurtByTimestamp:0,HurtTime:0s,Inventory:[],Invulnerable:0b,LeftHanded:0b,Motion:[0.0d,-0.0784000015258789d,0.0d],Offers:{Recipes:[{buy:{Count:4b,id:"minecraft:emerald"},buyB:{Count:1b,id:"minecraft:air"},demand:0,maxUses:5,priceMultiplier:0.05f,rewardExp:1b,sell:{Count:1b,id:"minecraft:slime_ball"},specialPrice:0,uses:0,xp:1},{buy:{Count:2b,id:"minecraft:emerald"},buyB:{Count:1b,id:"minecraft:air"},demand:0,maxUses:5,priceMultiplier:0.05f,rewardExp:1b,sell:{Count:1b,id:"minecraft:glowstone"},specialPrice:0,uses:0,xp:1},{buy:{Count:1b,id:"minecraft:emerald"},buyB:{Count:1b,id:"minecraft:air"},demand:0,maxUses:12,priceMultiplier:0.05f,rewardExp:1b,sell:{Count:3b,id:"minecraft:yellow_dye"},specialPrice:0,uses:0,xp:1},{buy:{Count:1b,id:"minecraft:emerald"},buyB:{Count:1b,id:"minecraft:air"},demand:0,maxUses:5,priceMultiplier:0.05f,rewardExp:1b,sell:{Count:2b,id:"minecraft:lily_pad"},specialPrice:0,uses:0,xp:1},{buy:{Count:1b,id:"minecraft:emerald"},buyB:{Count:1b,id:"minecraft:air"},demand:0,maxUses:12,priceMultiplier:0.05f,rewardExp:1b,sell:{Count:1b,id:"minecraft:red_tulip"},specialPrice:0,uses:0,xp:1},{buy:{Count:5b,id:"minecraft:emerald"},buyB:{Count:1b,id:"minecraft:air"},demand:0,maxUses:4,priceMultiplier:0.05f,rewardExp:1b,sell:{Count:1b,id:"minecraft:tropical_fish_bucket"},specialPrice:0,uses:0,xp:1},{buy:{Count:10b,id:"minecraft:emerald"},buyB:{Count:5b,id:"minecraft:diamond"},demand:0,maxUses:2,priceMultiplier:0.05f,rewardExp:1b,sell:{Count:1b,id:"companion_bats:spirit_shard"},specialPrice:0,uses:0,xp:1}]},OnGround:1b,PersistenceRequired:0b,PortalCooldown:0,Pos:[-83.5d,68.0d,14.5d],Rotation:[-63.150177f,0.0f],Tags:["collective-fabric.checked"],UUID:[I;-1265376492,-447396062,-1777166456,701870280],bm_IsStunted:0b,cardinal_components:{"apoli:powers":{Powers:[]}}}
Stacktrace:
	at net.minecraft.class_1937.method_18472(class_1937.java:487)
	at net.minecraft.class_3218.method_31420(class_3218.java:360)
	at net.minecraft.class_5574.method_31791(class_5574.java:54)
	at net.minecraft.class_3218.method_18765(class_3218.java:328)
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:879)
	at net.minecraft.class_3176.method_3813(class_3176.java:324)
	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:823)
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:684)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:270)
	at java.base/java.lang.Thread.run(Thread.java:833)

-- Affected level --
Details:
	All players: 2 total; [class_3222['Celophy'/219, l='ServerLevel[world]', x=-82.30, y=68.00, z=11.30], class_3222['TragicalWind'/548, l='ServerLevel[world]', x=-132.90, y=63.00, z=393.42]]
	Chunk stats: 3652
	Level dimension: minecraft:overworld
	Level spawn location: World: (-75,72,12), Section: (at 5,8,12 in -5,4,0; chunk contains blocks -80,-64,0 to -65,319,15), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,-64,0 to -1,319,511)
	Level time: 2459568 game time, 233522 day time
	Level name: world
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Known server brands: fabric
	Level was modded: true
	Level storage version: 0x04ABD - Anvil
Stacktrace:
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:879)
	at net.minecraft.class_3176.method_3813(class_3176.java:324)
	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:823)
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:684)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:270)
	at java.base/java.lang.Thread.run(Thread.java:833)

-- System Details --
Details:
	Minecraft Version: 1.18.1
	Minecraft Version ID: 1.18.1
	Operating System: Linux (amd64) version 4.19.0-12-amd64
	Java Version: 17.0.1, Oracle Corporation
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode, sharing), Oracle Corporation
	Memory: 568757048 bytes (542 MiB) / 2957262848 bytes (2820 MiB) up to 31894405120 bytes (30416 MiB)
	CPUs: 24
	Processor Vendor: GenuineIntel
	Processor Name: Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz
	Identifier: Intel64 Family 6 Model 44 Stepping 2
	Microarchitecture: Westmere (Server)
	Frequency (GHz): 2.93
	Number of physical packages: 2
	Number of physical CPUs: 12
	Number of logical CPUs: 24
	Graphics card #0 name: ASPEED Graphics Family
	Graphics card #0 vendor: ASPEED Technology, Inc. (0x1a03)
	Graphics card #0 VRAM (MB): 0.00
	Graphics card #0 deviceId: 0x2000
	Graphics card #0 versionInfo: unknown
	Virtual memory max (MB): 77100.75
	Virtual memory used (MB): 124874.13
	Swap memory total (MB): 4574.00
	Swap memory used (MB): 2830.81
	JVM Flags: 15 total; -Xms512M -XX:MetaspaceSize=384M -XX:MaxMetaspaceSize=768M -XX:+UseSerialGC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=200 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:InitiatingHeapOccupancyPercent=15 -XX:SurvivorRatio=32 -XX:+PerfDisableSharedMem -XX:MaxTenuringThreshold=1 -XX:UseSSE=4 -XX:SoftMaxHeapSize=512M -XX:InitiatingHeapOccupancyPercent=0
	Fabric Mods: 
		adorn: Adorn 3.3.1+1.18.1
		adventurez: AdventureZ 1.4.7
		amethystequipment: Amethyst Equipment 1.18.1
		apoli: Apoli 2.2.2
		architectury: Architectury 3.7.20
		basicshields: Basic Shields [Fabric] 1.2.1-1.18.1
		bclib: BCLib 1.3.4
		betterend: Better End 1.0.2
		betternether: Better Nether 6.0.10
		biomemakeover: Biome Makeover 1.18.1-1.4.7
		boatcontainer: BoatContainer 1.0.4
		bountiful: Bountiful 2.0.1
		calio: Calio 1.4.2
		cardinal-components-base: Cardinal Components API (base) 4.0.0
		cardinal-components-entity: Cardinal Components API (entities) 4.0.0
		chas: Craftable Horse Armour & Saddle 1.18-1.8-Fabric
		chunky: Chunky 1.2.164
		cloth-api: Cloth API 3.1.61
		cloth-basic-math: cloth-basic-math 0.6.0
		cloth-client-events-v0: Cloth Client Events v0 3.1.61
		cloth-common-events-v1: Cloth Common Events v1 3.1.61
		cloth-config: Cloth Config v6 6.2.57
		cloth-datagen-api-v1: Cloth Datagen v1 3.1.61
		cloth-utils-v1: Cloth Utils v1 3.1.61
		collective-fabric: Collective (Fabric) 4.8
		com_electronwill_night-config_core: core 3.6.3
		com_electronwill_night-config_toml: toml 3.6.3
		com_eliotlash_mclib_mclib: mclib 18
		com_eliotlash_molang_molang: molang 18
		companion_bats: Companion Bats 1.19.0
		copperequipment: Copper Equipment fabric-1.18.1-v2.0.0
		creeperoverhaul: Creeper Overhaul 0.3.0
		crowdin-translate: CrowdinTranslate 1.4+1.18
		direbats: Direbats 2.1.0
		doubledoors-fabric: Double Doors (Fabric) 3.1
		dual_riders: DualRiders 1.0.5
		easymagic: Easy Magic 3.2.1
		emeraldequipment: Emerald Equipment 1.18.1
		enderitemod: Enderite Mod 1.3.3
		enderitestorage: Enderite Reinforced Storage 1.0.0-beta.1+1.18
		fabric: Fabric API 0.46.4+1.18
		fabric-api-base: Fabric API Base 0.4.2+d7c144a865
		fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.5.3+d7c144a865
		fabric-biome-api-v1: Fabric Biome API (v1) 6.0.2+d7c144a865
		fabric-command-api-v1: Fabric Command API (v1) 1.1.7+d7c144a865
		fabric-commands-v0: Fabric Commands (v0) 0.2.6+b4f4f6cd65
		fabric-containers-v0: Fabric Containers (v0) 0.1.19+d7c144a865
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.4.8+d7c144a865
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.9+3ac43d9565
		fabric-dimensions-v1: Fabric Dimensions API (v1) 2.1.10+a1d9bbf565
		fabric-entity-events-v1: Fabric Entity Events (v1) 1.4.6+d7c144a865
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.17+d7c144a865
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.9+d7c144a865
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.11+d7c144a865
		fabric-item-api-v1: Fabric Item API (v1) 1.3.1+691a79b565
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.3.6+3ac43d9565
		fabric-language-kotlin: Fabric Language Kotlin 1.7.1+kotlin.1.6.10
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.4.13+713c266865
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.9+d7c144a865
		fabric-mining-level-api-v1: Fabric Mining Level API (v1) 1.0.6+d7c144a865
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.11+b4f4f6cd65
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.19+d7c144a865
		fabric-networking-v0: Fabric Networking (v0) 0.3.6+b4f4f6cd65
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.11.4+d7c144a865
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.12+d7c144a865
		fabric-particles-v1: Fabric Particles (v1) 0.2.10+526dc1ac65
		fabric-permissions-api-v0: fabric-permissions-api 0.1-SNAPSHOT
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.9.2+ad01bfbd65
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.5+d7c144a865
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.1.19+3ac43d9565
		fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 1.0.2+d7c144a865
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.4.14+713c266865
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.12+d7c144a865
		fabric-structure-api-v1: Fabric Structure API (v1) 2.1.3+d7c144a865
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.2.8+d7c144a865
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.3.8+fb3b57b465
		fabric-transfer-api-v1: Fabric Transfer API (v1) 1.5.9+d7c144a865
		fabricloader: Fabric Loader 0.13.2
		fabricshieldlib: Fabric Shield Lib 1.5.1-pre1-1.18
		feature_nbt_deadlock_be_gone: Feature NBT Deadlock Be Gone - Fabric 1.0.1+1.18.1
		fiber: fiber 0.23.0-2
		flonters: Flonters 1.3.0+1.18.1
		flytre_lib: FlytreLib 1.7.1
		forgeconfigapiport: Forge Config API Port 3.1.1
		geckolib3: Geckolib 3.0.32
		glassdoor: Glass Doors 2.1.0+1.18
		graveyard: The Graveyard 1.5
		guardvillagers: Guard Villagers Fabric 1.0.7
		hologram-api: Hologram API 0.2.1+1.18-pre5
		hookshot: Hookshot 24.0
		hplus: Hopper Plus 3.1.2
		impaled: Impaled 1.0.2
		java: OpenJDK 64-Bit Server VM 17
		kambrik: Kambrik 3.0.1-1.18
		libcd: LibCapableData 3.0.3+1.16.3
		lithium: Lithium 0.7.7
		magna: Magna 1.7.0-1.18-pre1
		mannequins: Mannequins 2.0.0
		maybe-data: Maybe data 1.0.1-1.17
		medievalweapons: Medieval Weapons 1.1.7
		megaparrot: Megaparrot 1.0.3
		minecraft: Minecraft 1.18.1
		mm: Manningham Mills 2.3
		moretags: More Tags 2.0.3
		morevillagers-fabric: MoreVillagersFabric 3.1.1-SNAPSHOT
		no_null_processors: No Null Processors - Fabric 2.0.1+1.18.1
		notenoughcrashes: Not Enough Crashes 4.1.4+1.18.1
		obsidianequipment: Obsidian Equipment 1.18.1
		omega-config: OmegaConfig 1.2.3-1.18.1
		org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 1.6.10
		org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 1.6.10
		org_jetbrains_kotlin_kotlin-stdlib-jdk7: kotlin-stdlib-jdk7 1.6.10
		org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 1.6.10
		org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm: kotlinx-coroutines-core-jvm 1.5.2
		org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.5.2
		org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm: kotlinx-serialization-cbor-jvm 1.3.1
		org_jetbrains_kotlinx_kotlinx-serialization-core-jvm: kotlinx-serialization-core-jvm 1.3.1
		org_jetbrains_kotlinx_kotlinx-serialization-json-jvm: kotlinx-serialization-json-jvm 1.3.1
		origins: Origins 1.3.1
		origins-classes: Origins: Classes 1.2.4
		oxidized: Oxidized 1.6.1
		packet_tweaker: Packet Tweaker 0.2.0+1.18-pre1
		phosphor: Phosphor 0.8.1
		placeholder-api: Placeholder API 1.1.3+1.17.1
		playerabilitylib: Pal 1.3.0
		pollen: Pollen 1.0.0
		polymer: Polymer (Core) 0.2.0-beta.24+1.18.1
		polymer-legacy: Polymer (Compatibility) 0.1.9+0.2.0-beta.24+1.18.1
		pugh_tools: Pugh Tools 1.1.17
		puzzleslib: Puzzles Lib 3.1.5
		reach-entity-attributes: Reach Entity Attributes 2.1.1
		reinfbarrel: Reinforced Barrels 2.0.0-beta.1+1.18
		reinfchest: Reinforced Chests 2.0.0-beta.2+1.18
		reinfcore: Reinforced Core 2.0.0-beta.1+1.18
		reinfshulker: Reinforced Shulker Boxes 2.0.0-beta.2+1.18
		repurposed_structures: Repurposed Structures 3.3.12+1.18.1
		sandwichable: Sandwichable 1.2-rc4+1.18
		server_translations_api: Server Translations API 1.4.8+1.18-pre1
		sgui: sgui 1.0.0-rc6+1.18-pre5
		simple_backpack: SimpleBackpack 1.2.30
		skeletonhorsespawn-fabric: Skeleton Horse Spawn (Fabric) 1.9
		static-content: Static Content 1.0.2-1.16.2
		staticdata: Static Data 1.1.2
		stoneholm: Stoneholm 1.4.2
		subterrestrial: Subterrestrial 1.0.1
		totw_additions: Towers of the Wild: Additions 1.1.0
		towers_of_the_wild_reworked: Towers of the Wild: Reworked 2.1.1
		trapexpansion: Trap Expansion 1.3.3
		treeaxe: The TreeAxe 1.1.32
		twigs: Twigs 1.1.2
		universal-graves: Universal Graves 2.0.0-rc.4+1.18.1
		vanilla-hammers: Vanilla Hammers 3.2.0-1.18.1
		vanillaexcavators: Vanilla Excavators 1.1.0
		voicechat: Simple Voice Chat 1.18.1-2.2.16
		wildworld: Wild World 3.1.0+1.18
		wolveswitharmor: Wolves With Armor 1.7.0-1.18
		zombiehorsespawn-fabric: Zombie Horse Spawn (Fabric) 3.0
	Server Running: true
	Player Count: 2 / 100; [class_3222['Celophy'/219, l='ServerLevel[world]', x=-82.30, y=68.00, z=11.30], class_3222['TragicalWind'/548, l='ServerLevel[world]', x=-132.90, y=63.00, z=393.42]]
	Data Packs: vanilla, Fabric Mods
	Is Modded: Definitely; Server brand changed to 'fabric'
	Type: Dedicated Server (map_server.txt)
	Suspected Mods: Minecraft (minecraft)

**Comment by warjort** [comment]
````
java.lang.ClassCastException: class net.minecraft.class_3989 cannot be cast to class net.minecraft.class_1657 (net.minecraft.class_3989 and net.minecraft.class_1657 are in unnamed module of loader net.fabricmc.loader.impl.launch.knot.KnotClassLoader @5f341870)
at net.minecraft.item.ItemStack.handler$znb000$finishUsingItem(ItemStack:1141)
````

class_3989 = WanderTraderEntity
class_1657 = PlayerEntity

You have a mod using a mixin for an ItemStack that thinks a wandering trader is a player.
Or probably more accurately, it falsely assumes that only players use items.

Whoever wrote this mod didn't include their mod name in the mixin name and just called it "finishUsingItem".
So we can only guess which mod it is. 
The usual way is remove mods in a test environment until the find the problem mod, but that is a difficult approach if you can't reproduce the circumstances of the crash.

**Comment by celophys** [comment]
Thanks for your help, i will try it!

**Comment by warjort** [comment]
This says it is in the ````collective-fabric```` mod and fixed in version 4.9 (you have 4.8)
https://github.com/ricksouth/serilum-mc-mods/issues/950
